tests/plugins/test_virtual_packages.py

```diff
@@ -92,71 +98,113 @@ def test_cuda_override_none(clear_cuda_version):
+    monkeypatch.setenv("CONDA_OVERRIDE_ARCHSPEC", version or "")
+    reset_context()
+    assert any(prec.name == "__archspec" for prec in get_virtual_precs()) is expected
+
+
+@pytest.mark.parametrize("version,expected", [(None, True), ("1.0", True)])
+def test_linux_override(monkeypatch: MonkeyPatch, version: str | None, expected: bool):
+    """Conda will still produce a linux virtual package when CONDA_OVERRIDE_LINUX=""."""
+    monkeypatch.setenv("CONDA_SUBDIR", "linux-64")
+    monkeypatch.setenv("CONDA_OVERRIDE_LINUX", version or "")
+    reset_context()
+    assert context.subdir == "linux-64"
+    assert any(prec.name == "__linux" for prec in get_virtual_precs()) is expected
+
+
+@pytest.mark.parametrize("version,expected", [(None, False), ("1.0", True)])
+def test_glibc_override(monkeypatch: MonkeyPatch, version: str | None, expected: bool):
     """Conda should not produce a libc virtual package when CONDA_OVERRIDE_GLIBC=""."""
-    for version in "", "1.0":
-        with env_vars(
-            {"CONDA_SUBDIR": "linux-64", "CONDA_OVERRIDE_GLIBC": version},
-            stack_callback=conda_tests_ctxt_mgmt_def_pol,
-        ):
-            packages = conda.core.index.get_reduced_index(
-                context.default_prefix,
-                context.default_channels,
-                context.subdirs,
-                (),
-                context.repodata_fns[0],
-            )
-            virtual = [p for p in packages if p.channel.name == "@"]
-            libc_exported = any("libc" in p.name for p in virtual)
-            assert libc_exported == bool(version)
-
-
-def test_osx_override():
+    monkeypatch.setenv("CONDA_SUBDIR", "linux-64")
+    monkeypatch.setenv("CONDA_OVERRIDE_GLIBC", version or "")
+    reset_context()
+    assert context.subdir == "linux-64"
+    assert any(prec.name == "__glibc" for prec in get_virtual_precs()) == expected
+
+
+@pytest.mark.parametrize("version,expected", [(None, False), ("1.0", True)])
+def test_osx_override(monkeypatch: MonkeyPatch, version: str | None, expected: bool):
     """Conda should not produce a osx virtual package when CONDA_OVERRIDE_OSX=""."""
-    for version in "", "1.0":
-        with env_vars(
-            {"CONDA_SUBDIR": "osx-64", "CONDA_OVERRIDE_OSX": version},
-            stack_callback=conda_tests_ctxt_mgmt_def_pol,
-        ):
-            packages = conda.core.index.get_reduced_index(
-                context.default_prefix,
-                context.default_channels,
-                context.subdirs,
-                (),
-                context.repodata_fns[0],
-            )
-            virtual = [p for p in packages if p.channel.name == "@"]
-            osx_exported = any("osx" in p.name for p in virtual)
-            assert osx_exported == bool(version)
+    monkeypatch.setenv("CONDA_SUBDIR", "osx-64")
+    monkeypatch.setenv("CONDA_OVERRIDE_OSX", version or "")
+    reset_context()
```

Comment:
Won't this leave the `context` object in a modified state after the test is done? Or is there a fixture reverting that silently?

Reply:
technically yes it will remain in a modified state, in most cases this wont be an issue (`conda_cli` for example ends up effectively doing a `reset_context`)

Reply:
I'm just concerned this ends up being a corner case where we leak state into other tests that are not doing any monkey patching or context resetting, and given our "split-by-duration" logic, test ordering is shuffled all the time. So we might even run things side by side with other module. It's just safer to restore the context once done.